src/leddrv.c

```diff
@@ -3,46 +3,61 @@
 #define LED_DRIVE_STRENTH GPIO_ModeOut_PP_20mA
 #define LED_PINCOUNT (23)
 
+#define PA_BASE 	((uint32_t *)0x400010A0)
```

Comment:
I don't know if we are going to access non 4 bytes aligned offsets, but in that case, it will deref the wrong address

Reply:
There is no access to 4 bytes aligned offsets. But I've just found pre-defined base address in the StdPeriphDriver; I'll use that instead.

Reply:
I see you just changed the code, so I guess you understood the issue with gpio_base being already uint32_t aligned

`*(uint32_t *)(gpio_base + GPIO_PD_DRV) = *cfg;`

Reply:
I had a workaround for that before the last change:
```
uint32_t b = (uint32_t)gpio_base; // <--here
*(uint32_t *)(b + GPIO_PD_DRV) = *cfg;
```
It seems kind of ugly but I changed to use pre-defined base address in the last change anyway.